RecoDataProducts/inc/KalSeed.hh

```diff
@@ -53,7 +53,8 @@ namespace mu2e {
     KinKal::TimeRange timeRange() const { return KinKal::TimeRange(_segments.front()._tmin,_segments.back()._tmax); }
     bool hasCaloCluster() const { return _chit.caloCluster().isNonnull(); }
     art::Ptr<CaloCluster> const& caloCluster() const { return _chit.caloCluster(); }
-    std::vector<KalSegment>::const_iterator nearestSeg(double time)  const;
+    std::vector<KalSegment>::const_iterator nearestSegment(double time)  const;
+    std::vector<KalSegment>::const_iterator t0Segment(double& t0) const; // return the segment associated with the t0 value, or the closest to it.  Also return the t0 value
```

Comment:
Have you considered returning a small struct that hods the iterator and t0?  It's more obvious to a new user than a return argument.  And I expect that this function will be used by new users.  My suggestion is slightly more expensive than yours but I doubt that's important in this context.